src/main/java/org/isf/opd/manager/OpdBrowserManager.java

```diff
@@ -150,6 +151,27 @@ public List<Opd> getOpd(boolean oneWeek) throws OHServiceException {
 		return ioOperations.getOpdList(oneWeek);
 	}
 	
+	/**
+	 * Return all Opds within specified dates and parameters
+	 * 
+	 * @param ward 
+	 * @param diseaseTypeCode
+	 * @param diseaseCode
+	 * @param dateFrom
+	 * @param dateTo
+	 * @param ageFrom
+	 * @param ageTo
+	 * @param sex
+	 * @param newPatient
+	 * @param patientCode
+	 * @return the list of Opds. It could be <code>null</code>.
+	 * @throws OHServiceException
+	 */
+	public List<Opd> getOpd(Ward ward, String diseaseTypeCode, String diseaseCode, LocalDateTime dateFrom, LocalDateTime dateTo, int ageFrom, int ageTo, char sex, char newPatient, int patientCode)
+			throws OHServiceException {
+		return ioOperations.getOpdList2(ward, diseaseTypeCode, diseaseCode, dateFrom, dateTo, ageFrom, ageTo, sex, newPatient, patientCode);
+	}
+
```

Comment:
this method has been already removed because if one wants the OPD list by patientCode the method `getOpdList(int patientcode)` is already there for this purpose

Reply:
the front need to get Opd by patientCode on the range define by dateFrom and dateTo. this method is use on the front. that is why i have add its

Reply:
as said on [Slack](https://join.slack.com/t/openhospitalworkspace/shared_invite/zt-cg9iu1d0-sjTuWWziFgcDqiTWyMJBAw), after we can change this, but really one patient has maybe 10 visits in his/her life (40 if he/she is chronic, always in the same ward) doesn't really need to apply all those filters when there is a patient code, or at least not in this way (i.e. duplicating a lot of code, bad approach).

But if you really want to add those filters we will think about a better solution.
Thanks!